src/main/java/duke/validation/TaskValidator.java

```diff
@@ -0,0 +1,143 @@
+
+package duke.validation;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+import duke.exceptions.DukeException;
+
+public class TaskValidator {
+
+    /**
+     * Validate user input date and converts string to LocalDate if true
+     *
+     * @param date the string date input from user
+     * @throws DukeException if date input from user is invalid
+     */
+    public static LocalDate convertDate(final String date) throws DukeException {
+        boolean valid = false;
```

Comment:
Should this valid = false be deleted?  It seems that it is not used in this method.😅